### PR TITLE
[WFLY-11697] WS integration with Elytron on the client side

### PR DESCRIPTION
**Pull request by Skyllarr**
JIRA: 
https://issues.jboss.org/browse/WFLY-11697

### CODE REVIEW

File: elytron/WFLY-11697_WS_REST_client_integration_with_Elytron.adoc [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+If wildfly-config.xml file is present when using RESTEasy or Web Services on the client side, this authentication configuration will be parsed and made available for this client.
+
+=== Nice-to-Have Requirements
+
+=== Non-Requirements
+
+== Implementation Plan
+
+SPI will be exposed in RESTEasy and Web Services for client's authentication configuration. This SPI will be implemented in Elytron.
+
```
Comment (Skyllarr) [inline elytron/WFLY-11697_WS_REST_client_integration_with_Elytron.adoc]: Alternatively, this SPI will be in its own module that both RestEasy and WS depend on and Elytron will implement.

Comment (Skyllarr) [reply elytron/WFLY-11697_WS_REST_client_integration_with_Elytron.adoc]: @asoldano So RestEasy will have SPI exposed that Elytron client implements and the loading of service providers could happen during build of RestEasyClientBuilder. Does this sound reasonable?

Comment (asoldano) [reply elytron/WFLY-11697_WS_REST_client_integration_with_Elytron.adoc]: @Skyllarr I suggest dealing with JBossWS and RESTEasy independently. JBossWS has its own SPI component, jbossws-spi (https://github.com/jbossws/jbossws-spi), which really looks like where the additional spi should be defined.
RESTEasy doesn't have such clearly defined spi component, so I believe a new module will have to be added to the project.

Comment (Skyllarr) [reply elytron/WFLY-11697_WS_REST_client_integration_with_Elytron.adoc]: @asoldano Thank you for your reply! We will split this issue and will deal with JBossWS and RESTEasy separately.

**Comment by Skyllarr** [comment]
Hello @asoldano . This PR is now only for the WS side. I force-pushed new analysis document. Can you please take a look? Thank you in advance.

**Comment by asoldano** [comment]
Thanks @Skyllarr . I'm adding @jimma to the discussion here, as he's now leading JBossWS / WebServices.
Jim, Diana here is designing the Elytron-WS integration that will have soon to be implemented. Please check if the proposal makes sense.

**Comment by jimma** [comment]
@Skyllarr  @asoldano  Is there background thing fot me to understand the purpose of add wildfly-config.xml in ws client ?

**Comment by Skyllarr** [comment]
Hello @jimma. This proposal is about being able to make use of elytron on the client side of the web services deployment. [Wildfly-config.xml](https://docs.jboss.org/author/display/WFLY/WildFly+Client+Configuration) contains authentication configuration for the client to use.

**Comment by darranl** [comment]
The backround really is the WildFly Elytron authentication client is already used by a number of our clients allowing for both use of an API and pre-defined configurations.  As well as configuration this set of APIs is also fundamental for features such as identity propagation.  

At the moment WS and Rest clients are not supporting the common framework available within the application server.

File: elytron/WFLY-11697_WS_client_integration_with_Elytron.adoc [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+The method setConfigHandlers takes client config file as a parameter.
+This file adheres to the schema
+http://anonsvn.jboss.org/repos/jbossws/spi/tags/jbossws-spi-2.1.0.Beta1/src/main/resources/schema/jbossws-jaxws-config_4_0.xsd[jbossws-jaxws-config schema] defined in jbossws-spi.
+Possible solutions are:
+
+* schema of wildfly-config.xml file and jaxws-config file could be combined and user can specify everything in single file
+* these configs will stay separate and user will provide paths to both files (if some information is conflicting, wildfly-config.xml setting takes precedence?)
+* the existence of wildfly-config.xml and its processing will be done by default in setConfigHandlers method
+* new method for this config file will be added (the use could look like this: `configurerBuilder.setConfigHandlers().setAuthHandlers().build()`)
```
Comment (Skyllarr) [inline elytron/WFLY-11697_WS_client_integration_with_Elytron.adoc]: @darranl @jimma I am guessing the last two options are the most viable? We can schedule a call if preferred. Thanks!

Comment (darranl) [reply elytron/WFLY-11697_WS_client_integration_with_Elytron.adoc]: jaxws-config could also make use of the wildfly-config.xml - but I think for this RFE the priority is we can already define client side authentication policies in the wildfly-config.xml so find a way to make use of them.

File: elytron/WFLY-11697_WS_client_integration_with_Elytron.adoc [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+N/A
+
+== Requirements
+
+=== Hard Requirements
+
+The WS-Security standard offers three methods of authentication. Those methods are:
+
+* UsernameToken Profile (adds credentials to header in the SOAP requests)
+* X 508 Certificates Token Profile (to use with TLS protocol)
```
Comment (jbliznak) [inline elytron/WFLY-11697_WS_client_integration_with_Elytron.adoc]: Minor one - should this be X.509?

Comment (Skyllarr) [reply elytron/WFLY-11697_WS_client_integration_with_Elytron.adoc]: Fixed.

**Pull request merged**